### PR TITLE
Update lucene to version 8.11.2

### PR DESCRIPTION
**Pull request by tuomas2**
Replaces #15 

This gave access to some new features in Lucene, such as Regular Expression search. This is a major refactor because I updated Lucene 5 major versions.

I tested several languages, English, Czech, Chinese, Japanese, Thai and search works in these languages. I am not capable to test if the stemming is good for all languages, so some more testing by native speakers is necessary.

### CODE REVIEW

**Comment by tuomas2** [comment]
So summarizing @JJK96 , I would like that we try to:
- Remove AbstractBookAnalyzer alltogether, and all custom analyzers that are based on that.
- Use StopwordAnalyzer as a baseclass for our custom analyzers (KeyAnalyzer etc)
- Modify properties file / factory accordingly to use classes from core and other libs.
- Change filter classes (used by some analyzers like KeyAnalyzer) so that they do not store book (as it does not seem to be used)

**Comment by JJK96** [comment]
- [x] Remove AbstractBookAnalyzer alltogether, and all custom analyzers that are based on that.
- [ ] Use StopwordAnalyzer as a baseclass for our custom analyzers (KeyAnalyzer etc)
    * I used Analyzer as the base, since stopwording was not used by these classes.
- [x] Modify properties file / factory accordingly to use classes from core and other libs.
- [x] Change filter classes (used by some analyzers like KeyAnalyzer) so that they do not store book (as it does not seem to be used)

File: src/main/java/org/crosswire/jsword/index/lucene/LuceneIndex.java [hunk 1]
```diff
@@ -401,17 +415,23 @@ private void generateSearchIndexImpl(Progress job, List<Key> errors, IndexWriter
...
-        Field bodyStemField = new Field(FIELD_BODY_STEM, "", Field.Store.NO, Field.Index.ANALYZED, Field.TermVector.NO);
-        Field introField = new Field(FIELD_INTRO, "", Field.Store.YES, Field.Index.ANALYZED, Field.TermVector.NO);
-        Field introStemField = new Field(FIELD_INTRO_STEM, "", Field.Store.NO, Field.Index.ANALYZED, Field.TermVector.NO);
-        Field strongField = new Field(FIELD_STRONG, "", Field.Store.NO, Field.Index.ANALYZED, Field.TermVector.YES);
-        Field xrefField = new Field(FIELD_XREF, "", Field.Store.NO, Field.Index.ANALYZED, Field.TermVector.NO);
-        Field noteField = new Field(FIELD_NOTE, "", Field.Store.NO, Field.Index.ANALYZED, Field.TermVector.NO);
-        Field headingField = new Field(FIELD_HEADING, "", Field.Store.YES, Field.Index.ANALYZED, Field.TermVector.NO);
-        Field headingStemField = new Field(FIELD_HEADING_STEM, "", Field.Store.NO, Field.Index.ANALYZED, Field.TermVector.NO);
-        Field morphologyField  = new Field(FIELD_MORPHOLOGY , "", Field.Store.NO, Field.Index.ANALYZED, Field.TermVector.NO);
+        FieldType stored_not_analyzed = new FieldType(StringField.TYPE_STORED);
```
Comment (tuomas2) [inline src/main/java/org/crosswire/jsword/index/lucene/LuceneIndex.java:418]: camelCaps

File: src/main/java/org/crosswire/jsword/index/lucene/analysis/AnalyzerFactory.java [hunk 2]
```diff
@@ -44,32 +51,77 @@
...
  */
 public final class AnalyzerFactory {
-    public AbstractBookAnalyzer createAnalyzer(Book book) {
-        AbstractBookAnalyzer newObject = null;
-        Language lang = book == null ? null : book.getLanguage();
+    public Analyzer createAnalyzer(Book book) {
+        return createAnalyzer(book, false);
+    }
+
+    public Analyzer createAnalyzer(Book book, Boolean stopwording) {
```
Comment (tuomas2) [inline src/main/java/org/crosswire/jsword/index/lucene/analysis/AnalyzerFactory.java:58]: stopWording

File: src/main/java/org/crosswire/jsword/index/lucene/analysis/AnalyzerFactory.java [hunk 3]
```diff
@@ -44,32 +51,77 @@
...
+        Map<String, Analyzer> analyzerPerField = new HashMap<String, Analyzer>();
+
+        // Content is analyzed using natural language analyzer
+        // (stemming, stopword etc)
+        analyzerPerField.put(LuceneIndex.FIELD_BODY, analyzer);
+        analyzerPerField.put(LuceneIndex.FIELD_BODY_STEM, analyzer);
+        analyzerPerField.put(LuceneIndex.FIELD_INTRO_STEM, analyzer);
+        analyzerPerField.put(LuceneIndex.FIELD_HEADING_STEM, analyzer);
+        //analyzerPerField.put(LuceneIndex.FIELD_HEADING, myNaturalLanguageAnalyzer);  //heading to use same analyzer as BODY
+        //analyzerPerField.put(LuceneIndex.FIELD_INTRO, myNaturalLanguageAnalyzer);
```
Comment (tuomas2) [inline src/main/java/org/crosswire/jsword/index/lucene/analysis/AnalyzerFactory.java:110]: unnecessary comments?

File: src/main/resources/IndexMetadata.properties [hunk 4]
```diff
@@ -21,9 +21,7 @@
 # 1.1 : Added field = strong, heading, xref, note
 # 1.2 : content: Uses natural language analysis (Stemming, CJK tokenization) for 19 languages. English bibles use SimpleLuceneAnalyzer
 
-Latest.Index.Version=1.2
-Lucene.Version=3.0
-
+Latest.Index.Version=1.3
```
Comment (tuomas2) [inline src/main/resources/IndexMetadata.properties:24]: Should add comment above (as there seem to be version history)

**Comment by tuomas2** [comment]
I'll merge these both to develop and start preparing a beta release. Looks good so far, but haven't tested yet in practice.

**Pull request merged**